src/main/java/jenkins/scm/impl/SingleSCMSource.java

```diff
@@ -79,14 +78,30 @@
     /**
      * Our constructor.
      *
-     * @param id   source id.
      * @param name the name of the branch.
      * @param scm  the configuration.
+     * @since 2.2.0
      */
     @SuppressWarnings("unused") // stapler
     @DataBoundConstructor
+    public SingleSCMSource(String name, SCM scm) {
+        this.name = name;
+        this.scm = scm;
+    }
+
+    /**
+     * Legacy constructor.
+     *
+     * @param id   source id.
+     * @param name the name of the branch.
+     * @param scm  the configuration.
+     * @deprecated use {@link #SingleSCMSource(String, SCM)} and {@link #setId(String)}.
+     */
+    @Restricted(NoExternalUse.class)
```

Comment:
Why? If the old call still works and in the same way, this is just harassing plugin developers needlessly.